js/application/appInit.js

```diff
@@ -0,0 +1,270 @@
+      var direction = this.direction;
+      var dirNode = document.getElementsByTagName("html")[0];
+      domAttr.set(dirNode, "dir", direction);
+    },
+
+    _createWebmap: function (webmapItem) {
+      var webmap;
+
+      if (!webmapItem) {
+        var error = new Error("main:: webmap data does not exist.");
+        this.reportError(error);
+        return;
+      }
+
+      if (webmapItem.data) {
+        if (webmapItem.data instanceof Error) {
+          this.reportError(webmapItem.data);
+        }
+        else {
+          webmap = new WebMap({
+            portalItem: webmapItem.data
+          });
+        }
+      }
+      else if (webmapItem.json) {
+        webmap = WebMap.fromJSON(webmapItem.json.itemData);
+        webmap.portalItem = webmapItem.json.item;
+      }
+      if (webmap) {
+        var viewProperties = {
+          map: webmap,
+          container: "viewDiv"
+        };
+        if (!this.config.title && webmap.portalItem && webmap.portalItem.title) {
+          this.config.title = webmap.portalItem.title;
+        }
+        lang.mixin(viewProperties, this.urlParamHelper.getViewProperties(this.config));
+
+        var view = new MapView(viewProperties);
+        view.then(function (response) {
+          this.urlParamHelper.addToView(view, this.config);
+
+          domClass.remove(document.body, CSS.loading);
+          document.title = this.config.title;
+
+          // Launch the app specifics
+          this.main.runMap(this.config, webmap, view);
+        }.bind(this), this.reportError);
+      }
+    },
+
+    _createWebscene: function (websceneItem) {
+      var webscene;
+      if (!websceneItem) {
+        var error = new Error("main:: webscene data does not exist.");
+        this.reportError(error);
+        return;
+      }
+      if (websceneItem.data) {
+        if (websceneItem.data instanceof Error) {
+          this.reportError(websceneItem.data);
+        }
+        else {
+          webscene = new WebScene({
+            portalItem: websceneItem.data
+          });
+        }
+      }
+      else if (websceneItem.json) {
+        webscene = WebScene.fromJSON(websceneItem.json.itemData);
+        webscene.portalItem = websceneItem.json.item;
+      }
+      if (webscene) {
+        var viewProperties = {
+          map: webscene,
+          container: "viewDiv"
+        };
+
+        if (!this.config.title && webscene.portalItem && webscene.portalItem.title) {
+          this.config.title = webscene.portalItem.title;
+        }
+
+        lang.mixin(viewProperties, this.urlParamHelper.getViewProperties(this.config));
+
+        var view = new SceneView(viewProperties);
+
+        view.then(function (response) {
+
+          this.urlParamHelper.addToView(view, this.config);
+
+          domClass.remove(document.body, CSS.loading);
+          document.title = this.config.title;
+
+          // Launch the app specifics
+          this.main.runMap(this.config, webscene, view);
```

Comment:
I agree that there might be a better name than `runMap`.  I was trying to distinguish it from `runGroup`.

Because of inheritance, the same function can be used for 2D and 3D because WebMap and WebScene both inherit from Map (second argument) and MapView and SceneView both inherit from View (third argument).  This permits the boilerplate to have just two `run*` functions.  But that might be a false economy--perhaps it is clearer to have a run for 2D, a run for 3D, and a run for groups?